examples/templates/azure-linux/main.tf

```diff
@@ -11,64 +11,176 @@ terraform {
   }
 }
 
-variable "location" {
+data "coder_parameter" "location" {
+  name        = "Location"
   description = "What location should your workspace live in?"
   default     = "eastus"
-  validation {
-    condition = contains([
-      "eastus",
-      "southcentralus",
-      "westus2",
-      "australiaeast",
-      "southeastasia",
-      "northeurope",
-      "westeurope",
-      "centralindia",
-      "eastasia",
-      "japaneast",
-      "brazilsouth",
-      "asia",
-      "asiapacific",
-      "australia",
-      "brazil",
-      "india",
-      "japan",
-      "southafrica",
-      "switzerland",
-      "uae",
-    ], var.location)
-    error_message = "Invalid location!"
-  }
-}
-
-variable "instance_type" {
+  icon        = "/emojis/"
+  mutable     = false
+  option {
+    name  = "East US"
+    value = "eastus"
+    icon  = "/emojis/1f1fa-1f1f8.png"
+  }
+  option {
+    name  = "South Central US"
+    value = "southcentralus"
+    icon  = "/emojis/1f1fa-1f1f8.png"
+  }
+  option {
+    name  = "West US 2"
+    value = "westus2"
+    icon  = "/emojis/1f1fa-1f1f8.png"
+  }
+  option {
+    name  = "Australia East"
+    value = "australiaeast"
+    icon  = "/emojis/1f1e6-1f1fa.png"
+  }
+  option {
+    name  = "Southeast Asia"
+    value = "southeastasia"
+    icon  = "/emojis/1f1f8-1f1e6.png"
+  }
+  option {
+    name  = "North Europe"
+    value = "northeurope"
+    icon  = "/emojis/1f1f3-1f1ea.png"
+  }
+  option {
+    name  = "West Europe"
+    value = "westeurope"
+    icon  = "/emojis/1f1f3-1f1ea.png"
+  }
+  option {
+    name  = "Central India"
+    value = "centralindia"
+    icon  = "/emojis/1f1ee-1f1f3.png"
+  }
+  option {
+    name  = "East Asia"
+    value = "eastasia"
+    icon  = "/emojis/1f1f0-1f1f7.png"
+  }
+  option {
+    name  = "Japan East"
+    value = "japaneast"
+    icon  = "/emojis/1f1ef-1f1f5.png"
+  }
+  option {
+    name  = "Brazil South"
+    value = "brazilsouth"
+    icon  = "/emojis/1f1e7-1f1f7.png"
+  }
+  option {
+    name  = "Asia"
+    value = "asia"
+    icon  = "/emojis/1f1f0-1f1f7.png"
+  }
+  option {
+    name  = "Asia Pacific"
+    value = "asiapacific"
+    icon  = "/emojis/1f1f0-1f1f7.png"
+  }
+  option {
+    name  = "Australia"
+    value = "australia"
+    icon  = "/emojis/1f1e6-1f1fa.png"
+  }
+  option {
+    name  = "Brazil"
+    value = "brazil"
+    icon  = "/emojis/1f1e7-1f1f7.png"
+  }
+  option {
+    name  = "India"
+    value = "india"
+    icon  = "/emojis/1f1ee-1f1f3.png"
+  }
+  option {
+    name  = "Japan"
+    value = "japan"
+    icon  = "/emojis/1f1ef-1f1f5.png"
+  }
+  option {
+    name  = "South Africa"
+    value = "southafrica"
+    icon  = "/emojis/1f1ff-1f1e6.png"
+  }
+  option {
+    name  = "Switzerland"
+    value = "switzerland"
+    icon  = "/emojis/1f1e8-1f1ed.png"
+  }
+  option {
+    name  = "UAE"
```

Comment:
UAE North?

Reply:
why we need to sau UAE North? there is only 1 UAE.

Reply:
I'd like to see this list consistent with [Azure docs](https://learn.microsoft.com/en-us/azure/reliability/availability-zones-service-support).

Reply:
I replaced all directions with City Names? as per Azure docs. is it ok?